### DISCUSSION TITLE
How can i async await in setting fetchOptions in a createClient function?

### DISCUSSION THREAD

**Discussion by CrispenGari** [urql-graphql/urql#2975] [Q&A] [answered]
I want to fetch the `token`  from `async-storage` but it returns a `Promise` of type `string` or `undefined`.  Since `fetchOptions`  is a function that returns a object of type `RequestInit` here is what i've tried

```ts
export const client = createClient({
  url,
  fetchOptions: async () => {
    const token = (await retrieve(TOKEN_KEY)) ?? "";
    return {
      credentials: "include",
      headers: { authorization: token ? `Bearer ${token}` : "" },
    };
  },
});
```
But typescript is not happy with me. How can I wake around this??

**Comment by JoviDeCroock** [comment, marked as answer]
You can't, however the [authExchange](https://formidable.com/open-source/urql/docs/advanced/authentication/) or the [mapExchange](https://formidable.com/open-source/urql/docs/api/core/#mapexchange) can solve this for you

**Reply by CrispenGari** [reply]
Thanks @JoviDeCroock exactly what i was looking for 🎈🎈🎆
